pkg/oci/oci.go

```diff
@@ -102,7 +106,25 @@ var getLocalOciStore = sync.OnceValues(func() (*oci.Store, error) {
 	if err := os.MkdirAll(filepath.Dir(defaultOciStore), 0o700); err != nil {
 		return nil, err
 	}
-	return oci.New(defaultOciStore)
+
+	// calling oci.New causes it to read the index.json file, lock to avoid it
+	// messing up with other instances writing this
```

Comment:
This comment makes me unsure if this is truly concurrent. According to that comment `oci.New` is the point where the index is read. With the following codepath we get in troublt:

t1=thread 1, t2=thread2
1. local oci store is empty
2. t1 calls getLocalOciStore(), it thinks there are no images
3. t2 calls getLocalOciStore(), it thinks there are no images
4. t1 downloads image 1 and saves it to the index
5. t2 downloads image 2 and saves it to the index. Since it read the index before 4, it will overwrite the index and save only image 2

Reply:
Ah, you're totally right. I switched back to an approach that locks the operation from the beginning to the end (includes the time needed to pull the image). I was checking around other options but I don't see any **easy** solution. Pulling the image to a temporary store seems also difficult to handle. IMHO this solution is good enough for now.

Reply:
Now after I have written https://github.com/inspektor-gadget/inspektor-gadget/pull/3842#discussion_r1908496461 the above scenario only comes up with multiple processes, not threads

Reply:
I assumed you're referring to multiple processes in your comment above. You're right, the difficult part is to handle multiple processes, the sync.Once and the internal mutex used by oci.New() protects it on the same process.